tests/wpt/web-platform-tests/html/semantics/forms/textfieldselection/select-event.html

```diff
@@ -47,10 +47,10 @@
   }
 ];
 
-for (const el of els) {
+els.forEach((el) => {
   const elLabel = el.localName === "textarea" ? "textarea" : "input type " + el.type;
 
-  for (const action of actions) {
+  actions.forEach((action) => {
```

Comment:
Why the change?

Reply:
This is explained in the commit message:

> There is a test for the firing of the select event in
tests/wpt/web-platform-tests/html/semantics/forms/textfieldselection/select-event.html,
however the test did not run due to this syntax error:
>
> (pid:26017) "ERROR:script::dom::bindings::error: Error at http://web-platform.test:8000/html/semantics/forms/textfieldselection/select-event.html:50:11 missing = in const declaration"
>
> This happens due to the us of the "for (const foo of ...)" construct.
> Per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of
this should actually work, so it's somewhat unsatisfying to have to
change the test.

Definitely not ideal to have to change this, so I'm open to other suggestions! But without changing this, the changes here are not fully tested (specifically, setting `selectionDirection` will now call `set_selection_range` and fire the `select` event, and this test file verifies that behaviour).

Reply:
Does `let` instead of `const` work? If not, then this looks like it's a problem with our implementation, and not the tests.

Reply:
Our spidermonkey version doesn't support `for (const foo of bar)`. We modify tests to use `for (let foo of bar)` instead when we want to be able to run them.

Reply:
I think I tried `let` but was having trouble for some reason. I'll have another go.

Reply:
Right, with `let`, when the function gets executed `action` is always set to the last item in `actions`, which results in a call to `setRangeText`, which fails. This seems wrong: `let` seems to be behaving like `var`. Is this another difference in Servo's version of SpiderMonkey?

Reply:
Not one that I'm aware of! Want to make an isolated example demonstrating it and file a new issue?

Reply:
Sure: https://github.com/servo/servo/issues/19535

How should we proceed with this PR?

Reply:
My inclination would be to revert this change and mark them as expected failures.

Reply:
Or maybe put the modified version in `tests/wpt/mozilla/` for the time being? Seems a shame not to have it tested in any way, as then we won't be sure we're not introducing regressions.

Reply:
Though as far as I can see those tests (in `tests/wpt/mozilla/`) don't actually get run by the CI? So maybe that wouldn't particularly help.

Reply:
No, tests/wpt/mozilla/ definitely get run by CI. Adding a modified version to that directory is a good idea.

Reply:
> 15:24 &lt;annevk> nox: I think it’s considered acceptable to rewrite if not too much of a concession in readability

We could just use the `forEach` method in that test.

Reply:
@jdm are you happy to do as @nox suggests (if so, this can be merged as it is), or shall I add the modified test in tests/wpt/mozilla?